plugins/operation_history/plugin.cpp

```diff
@@ -94,9 +94,29 @@ namespace golos { namespace plugins { namespace operation_history {
 
         ~plugin_impl() = default;
 
+        void erase_old_blocks() {
+            uint32_t head_block = database.head_block_num();
+            if (history_blocks <= head_block) {
+                uint32_t need_block = head_block - history_blocks + 1;
+                const auto& idx = database.get_index<operation_index>().indices().get<by_location>();
+                auto it = idx.begin();
+                while (it not_eq idx.end()) {
+                    auto next_it = it;
+                    ++next_it;
+                    uint32_t block = it->block;
+                    applied_operation op(*it);
+                    if (block <= need_block) {
+                        database.remove(*it);
+                    }
+                    it = next_it;
+                }
```

Comment:
Currently this loop runs on whole history.
For example, if I want to record 10000 blocks, this loop will iterate all of them on each iteration, but will remove only first N records (depends on amount of operations in block).